mod-source-record-storage-server/src/main/java/org/folio/services/RecordServiceImpl.java

```diff
@@ -102,7 +102,7 @@ public RecordServiceImpl(final RecordDao recordDao) {
           return recordDao.getRecordByMatchedId(txQE, record.getMatchedId())
             .compose(optionalMatchedRecord -> optionalMatchedRecord
               .map(matchedRecord -> recordDao.saveUpdatedRecord(txQE, ensureRecordForeignKeys(record.withGeneration(generation)), matchedRecord.withState(Record.State.OLD)))
-              .orElse(recordDao.saveRecord(txQE, ensureRecordForeignKeys(record.withGeneration(generation)))));
+              .orElseGet(() -> recordDao.saveRecord(txQE, ensureRecordForeignKeys(record.withGeneration(generation)))));
```

Comment:
just curious, why this change?

Reply:
orElse calls the method and calculates the return value anyway, orElseGet lets us avoid this.

Reply:
the saveRecord() was being called, alone with saveUpdatedRecord(), even when a record was found by matchedId